### PR TITLE
Update README with deprecated warning

### PR DESCRIPTION
**Pull request by Raquel10-RevenueCat**
## Motivation / Description
Update README with deprecated warning

### CODE REVIEW

**Review by gragera — COMMENTED** [review]
I have some questions about the Swift code!

File: code_blocks/🔌 Integrations & Events/third-party-integrations/onesignal_14.swift [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        // prints out all properties for testing
+        print("OSUserChangedState: \n\(state.jsonRepresentation())")
+
+        let onesignalId = state.current.onesignalId
+        let externalId = state.current.externalId
+
+        // Confirm the `onesignalId` is for the expected user by also checking the associated `externalId`
+        // For example, if the device is offline, there may be cached user-related requests
+        if let onesignalId,
+           (externalId == myExpectedExternalId)
```
[the remark this answers is not in the document]
Comment (Raquel10-RevenueCat) [reply code_blocks/🔌 Integrations & Events/third-party-integrations/onesignal_14.swift:37]: Yeah, here is the message: https://revenuecat.slack.com/archives/C028MU6LFGS/p1709928333790059?thread_ts=1696255290.873889&cid=C028MU6LFGS

File: code_blocks/🔌 Integrations & Events/third-party-integrations/onesignal_14.swift [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+        OneSignal.initialize("<onesignal_app_id>", withLaunchOptions: launchOptions)
+        OneSignal.User.addObserver(self as OSUserStateObserver)
+
+        // If you've already set up OneSignal, then users should already have
+        // a OneSignal Id. We can check if it's available and send it to RevenueCat
+        if let onesignalId = OneSignal.User.onesignalId {
+            // It is recommended to confirm the associated `externalId` to ensure
+            // you are capturing the correct data for the user you expect
+            if (OneSignal.User.externalId == myExpectedExternalId) {
```
[the remark this answers is not in the document]
Comment (Raquel10-RevenueCat) [reply code_blocks/🔌 Integrations & Events/third-party-integrations/onesignal_14.swift:14]: Yeah good point, updated to placeholder format so it is more clear this is not defined

**Comment by codykerns** [comment]
@Raquel10-RevenueCat apologies, this repo is no longer used - can you make the change in https://github.com/RevenueCat/docusaurus?

**Review by gragera — APPROVED** [review]
Thanks for doing this

**Comment by Raquel10-RevenueCat** [comment]
> @Raquel10-RevenueCat apologies, this repo is no longer used - can you make the change in https://github.com/RevenueCat/docusaurus?

Done!

> @Raquel10-RevenueCat apologies, this repo is no longer used - can you make the change in https://github.com/RevenueCat/docusaurus?

🤦‍♀️ Sure, done! https://github.com/RevenueCat/docusaurus/pull/152

I'll repurpose this PR to update the README of the repo instead :) (I see you already updated the PR template, thanks!)

**Review by codykerns — APPROVED** [review]

**Pull request merged**